roman-r-m/joplin-inline-tags-plugin
# Support CodeMirror 6

# Summary

This pull request does two things:
1. Updates the plugin build script. This is required to `require` CodeMirror 6 packages.
    - The `webpack.config.js` file was updated to an unreleased version to support this. (See [the relevant section in Joplin's beta markdown editor plugin tutorial](https://joplinapp.org/help/api/tutorials/cm6_plugin#update-the-plugin-build-script)).
2. Adds support for the CodeMirror 6-based beta markdown editor.


# Notes

- This pull request does not commit generated updates to `package-lock.json` because such changes are difficult to review.
- Requires Joplin >= 2.14.16.
- This pull request is similar to https://github.com/roman-r-m/joplin-plugin-quick-links/pull/15.

# Testing

1. Enable the beta editor
2. Add an existing tag to a note using the plugin
    - Verify that text is preserved by default
3. Create a new tag using the plugin
    - Verify that text is still preserved
4. Disable "keep `#tag` text in the editor"
5. Add an existing tag to a note using the plugin
    - Verify that the completion text has been removed
6. Create a new tag using the plugin
    - Verify that the completion text has been removed
8. Create a code block
9. Try to link to a tag from the code block
    - Verify that the tag completion dialog doesn't appear.
10. Disable the beta editor and repeat steps 1-9.

This has been tested successfully on OpenSUSE Linux in Joplin 2.14.16 on commit 3286b5dfcdec78a90e96d54e8dd47a293532e7f5.

[Screen recording: Shows above steps](https://github.com/roman-r-m/joplin-inline-tags-plugin/assets/46334387/2997885d-ecc2-4463-8e15-f1ab99ce3d3c)
